cartridges/int_cloudinary/cartridge/templates/default/include/cloudinaryStaticLibs.isml

```diff
@@ -5,13 +5,24 @@
     <isif condition="${pdict.cloudinary}">
         <div class="cloudinary-data-container d-none visually-hidden" data-cloudinary="<isprint value="${JSON.stringify(pdict.cloudinary)}" encoding="htmlunquote" />"></div>
     <iselse/>
-        <div class="cloudinary-data-container d-none visually-hidden" data-cloudinary="${preferences.CLDCloudName}"></div> 
+        <div class="cloudinary-data-container d-none visually-hidden" data-cloudinary="${preferences.CLDCloudName}"></div>
     </isif>
     <script src="${URLUtils.staticURL('/js/cloudinaryWidgets.js')}"></script>
     <isif condition="${pdict.cloudinary && pdict.cloudinary.videoEnabled && pdict.cloudinary.videoPlayerEnabled}">
         <link href="${preferences.CLDVideoPlayerCSSURL}" rel="stylesheet"/>
-        <script src="${preferences.CLDVideoPlayerJSURL}"></script>
     </isif>
+    <isscript>
+        pdict.CLDVideoPlayerJSURL = preferences.CLDVideoPlayerJSURL;
+    </isscript>
+    <script>
+        var CLDVideoPlayerJSURL = JSON.parse('<isprint value="${JSON.stringify(pdict.CLDVideoPlayerJSURL)}" encoding="jsonvalue"/>');
+            const script = document.createElement('script');
+            script.src = CLDVideoPlayerJSURL;
+            document.body.appendChild(script);
+            script.onload = () => {
```

Comment:
why do we need this onload logic scripts? This creates complexity and delays in page response time.
can't we just assign window.cldVideoPlayer = cloudinary.Cloudinary; after a direct call to the shrinkWrap script?

Reply:
I tried removing. If we don't have the script.onload it assign  window.cldVideoPlayer = cloudinary.Cloudinary; before the script load and it doesn't work, we are creating realtime script tag in DOM and checking if it's loaded, I don't think so the script.onload will create delays.

Reply:
yes, it shouldn't cause a delay but it is bound to create a race condition - depends on the caching status of each script.
I created this demo - https://kwxsfq.csb.app/ - where window.cloudinaryVideoPlayer is accessible and I have access to both modules. Can you check it out and see where is the gap?

Reply:
I don't understand, we are creating the script tag element in javascript dynamically, and when the script tag loads we call the function, there is no means to call the function without having the script tag loaded in DOM. Let me know if we can jump on a call to discuss